pkg/controller/migration/convert/errors.go

```diff
@@ -0,0 +1,51 @@
+package convert
+
+import "fmt"
+
+// ErrIncompatibleCluster indicates that a config option was detected in the existing install
+// which Operator does not support.
+type ErrIncompatibleCluster struct {
+	// err describes the error
+	err string
+	// fix explains what the user can do, if anything, to continue the migration.
+	fix string
+	// component identifies which component caused the problem.
+	component string
+}
+
+func (e ErrIncompatibleCluster) Error() string {
+	return fmt.Sprintf("%s. To fix it, %s on %s", e.err, e.fix, e.component)
+}
+
+const (
+	ComponentCalicoNode      = "daemonset/calico-node"
+	ComponentKubeControllers = "deployment/calico-kube-controllers"
+	ComponentTypha           = "deployment/calico-typha"
+	ComponentCNIConfig       = "cni-config"
+	ComponentIPPools         = "ippools"
+)
+
+var (
+	// FixFileFeatureRequest is returned when configuration could be migrated in the future
+	FixFileFeatureRequest = "file a feature request to support clusters with your configuration"
```

Comment:
You've raised two good points here. let me share my thoughts for each.

1. `FixFileBugReport` is only used in one place in lieu of raising a panic() in the code. Basically, we conditionally parse CNI configuration early in the code, and then analyze that parsed CNI config later. FixFileBugReport is used if we're trying to analyze the parsed CNI config but it's nil. Maybe instead of FixFileBugReport, we just need to raise some kind of UnexpectedError.

1. FixImpossible is meant to indicate there is a technical blocker from migrating this cluster that cannot ever be resolved. FixFileFeatureRequest is meant for cases where there is no technical blocker, but we just haven't written that migration code yet. I see your point though, perhaps that distinction isn't useful to the user, and I bet there's a lot of cases where there isn't a technical blocker, but just an unreasonable amount of work preventing us from writing that code, and the distinction between those is such a fine line.

I guess I'll propose:
- update so that the `fix` field is optional
- remove FixImpossible and FixFileBugReport to just omit the fix.